test/scripts/tls_cipher_check.sh

```diff
@@ -0,0 +1,164 @@
+#!/usr/bin/env bash
+
+# Script to check TLS cipher suites used in FIPS vs non-FIPS mode
+# Usage: ./tls_cipher_check.sh [fips|non-fips] [host:port]
+
+set -e
+
+MODE="${1:-non-fips}"
+HOST="${2:-localhost:8080}"
+
+# FIPS-compliant cipher suites (TLS 1.2)
+FIPS_TLS12_CIPHERS=(
+    "ECDHE-ECDSA-AES128-GCM-SHA256"
+    "ECDHE-ECDSA-AES256-GCM-SHA384"
+    "ECDHE-RSA-AES128-GCM-SHA256"
+    "ECDHE-RSA-AES256-GCM-SHA384"
+)
+
+# FIPS-compliant cipher suites (TLS 1.3)
+FIPS_TLS13_CIPHERS=(
+    "TLS_AES_128_GCM_SHA256"
+    "TLS_AES_256_GCM_SHA384"
+)
+
+# Non-FIPS cipher suites (TLS 1.2)
+NON_FIPS_TLS12_CIPHERS=(
+    "ECDHE-ECDSA-CHACHA20-POLY1305"
+    "ECDHE-RSA-CHACHA20-POLY1305"
+)
+
+# Non-FIPS cipher suites (TLS 1.3)
+NON_FIPS_TLS13_CIPHERS=(
+    "TLS_CHACHA20_POLY1305_SHA256"
+)
```

Comment:
Similar to the comments for the source code change, it would be great to have a reference to the source of truth for these lists of ciphers.